extensions/sql/asset/sql_asset_index_loader_proposal.md

```diff
@@ -0,0 +1,186 @@
+    PRIMARY KEY (asset_id)
+);
+
+
+-- table: edc_asset_dataaddress
+CREATE TABLE IF NOT EXISTS edc_asset_dataaddress (
+    asset_id VARCHAR(255) NOT NULL,
+    properties VARCHAR(MAX) NOT NULL,
+    PRIMARY KEY (asset_id),
+    FOREIGN KEY (asset_id) REFERENCES edc_asset (asset_id) ON DELETE CASCADE
+);
+
+COMMENT ON COLUMN edc_asset_dataaddress.properties is 'DataAddress properties serialized as JSON';
+
+
+-- table: edc_asset_property
+CREATE TABLE IF NOT EXISTS edc_asset_property (
+    asset_id VARCHAR(255) NOT NULL,
+    property_name VARCHAR(255) NOT NULL,
+    property_string_value VARCHAR(MAX) DEFAULT NULL,
+    property_object_value VARCHAR(MAX) DEFAULT NULL,
+    PRIMARY KEY (asset_id, property_name),
+    FOREIGN KEY (asset_id) REFERENCES edc_asset (asset_id) ON DELETE CASCADE,
+    CONSTRAINT con_edc_asset_property_single_value CHECK (
+        (property_string_value IS NOT NULL AND property_object_value IS NULL)
+     OR (property_string_value IS NULL AND property_object_value IS NOT NULL)
+     OR (property_string_value IS NULL AND property_object_value IS NULL))
+);
+
+COMMENT ON COLUMN edc_asset_property.property_name IS 
+    'Asset property key';
+
+COMMENT ON COLUMN edc_asset_property.property_string_value IS  
+    'Asset property string value if property of type string';
+
+COMMENT ON COLUMN edc_asset_property.property_object_value IS 
+    'Asset property object value serialized as JSON if not of type string';
+
+COMMENT ON CONSTRAINT con_edc_asset_property_single_value IS  
+    'All value fields are NULL, otherwise exclusively one of the value columns is non-NULLable';
+
+CREATE INDEX IF NOT EXISTS idx_edc_asset_property_string_value 
+  ON edc_asset_property(property_name, property_string_value);
+```
+
+## DML - AssetLoader
+#### AssetLoader: `void accept(Asset asset, DataAddress dataAddress)`
+
+Given example Asset data
+
+<table>
+  <thead>
+    <tr>
+      <td><b>id</b></td>
+      <td><b>asset:prop:name</b></td>
+      <td><b>asset:prop:description</b></td>
+      <td><b>asset:prop:version</b></td>
+      <td><b>asset:prop:contenttype</b></td>
+      <td><b>data-address</b></td>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>784cdfde-a50c-11ec-b909-0242ac120002</td>
+      <td>Max Power</td>
+      <td>Homer to the Max</td>
+      <td><i>NULL</i></td>
+      <td><i>NULL</i></td>
+      <td>
+        <code>
+          {
+            "uri": "https://projects.eclipse.org/proposals/eclipse-dataspace-connector"
+          }
+        </code>
+      </td>
+    </tr>
+    <tr>
+      <td>80b69de0-a50c-11ec-b909-0242ac120002</td>
+      <td>Seneca</td>
+      <td>Seneca the Younger</td>
+      <td><i>v2</i></td>
+      <td><i>application/pdf</i></td>
+      <td>
+        <code>
+          {
+            "uri": "https://archive.org/download/adlucilium02sene/adlucilium02sene.pdf"
+          }
+        </code>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+would translate into following DML:
+
+```sql
+BEGIN;
```

Comment:
We won't have to express transactions explicitly. That what we have the `TransactionContext` (and related classes) for.